File: backend/src/main/java/by/fpmibsu/findafriend/controller/commands/users/UpdateUserHandler.java
package by.fpmibsu.findafriend.controller.commands.users;

import by.fpmibsu.findafriend.application.mediatr.RequestHandler;
import by.fpmibsu.findafriend.dataaccesslayer.user.UserDao;
import by.fpmibsu.findafriend.dataaccesslayer.validtokens.ValidTokensDao;
import by.fpmibsu.findafriend.services.PasswordHasher;

public class UpdateUserHandler extends RequestHandler<Boolean, UpdateUserCommand> {
    private final PasswordHasher passwordHasher;
    private final UserDao userDao;
    private final ValidTokensDao tokensDao;
    public UpdateUserHandler(PasswordHasher passwordHasher, UserDao userDao, ValidTokensDao tokensDao) {
        this.passwordHasher = passwordHasher;
        this.userDao = userDao;
        this.tokensDao = tokensDao;
    }

    @Override
    public Boolean handle(UpdateUserCommand request) {
        var user = userDao.getEntityById(request.userId);
        if (user == null) {
            return false;
        }

        boolean invalidated = false;
        if (request.name != null) {
            user.getContacts().setName(request.name);
        }
        if (request.surname != null) {
            user.getContacts().setSurname(request.surname);
        }
        if (request.email != null) {
            user.getContacts().setEmail(request.email);
        }
        if (request.phoneNumber != null) {
            user.getContacts().setPhoneNumber(request.phoneNumber);
        }

        if (request.login != null || request.password != null) {
            boolean passwordIsValid = passwordHasher.verifyPassword(user, request.providedPassword) != PasswordHasher.PasswordVerificationResult.FAILED;
            if (passwordIsValid && request.login != null) {
                invalidated |= !request.login.equals(user.getLogin());
                user.setLogin(request.login);
            }
            if (passwordIsValid && request.password != null) {
                var hashedPassword = passwordHasher.hashPassword(request.password);
                invalidated |= !user.getPassword().equals(hashedPassword);
                user.setPassword(hashedPassword);
            }
        }

        userDao.update(user);
        if (invalidated) {
            tokensDao.invalidateTokens(user.getId());
        }
        return invalidated;
    }
}

File: BackendNET/Backend/Commands/Users/UpdateUser/UpdateUserHandler.cs
using Backend.Entities;
using Backend.Repository;
using MediatR;
using Microsoft.AspNetCore.Identity;

namespace Backend.Commands.Users.UpdateUser;

public class UpdateUserHandler : IRequestHandler<UpdateUserCommand, bool>
{
    private readonly IUserRepository _userRepository;
    private readonly ITokenRepository _tokenRepository;
    private readonly IPasswordHasher<User> _passwordHasher;

    public UpdateUserHandler(IUserRepository userRepository, ITokenRepository tokenRepository, IPasswordHasher<User> passwordHasher)
    {
        _userRepository = userRepository;
        _tokenRepository = tokenRepository;
        _passwordHasher = passwordHasher;
    }

    public async Task<bool> Handle(UpdateUserCommand request, CancellationToken cancellationToken)
    {
        var instance = await _userRepository.GetByIdAsync(request.UserId, cancellationToken);
        if (instance is null)
        {
            return false;
        }

        bool invalidated = request.Login is not null || request.Password is not null;
        instance.Name = request.Name ?? instance.Name;
        instance.Surname = request.Surname ?? instance.Surname;
        instance.Email = request.Email ?? instance.Email;
        instance.PhoneNumber = request.PhoneNumber ?? instance.PhoneNumber;
        // instance.Login = request.Login ?? instance.Login;
        if (request.Password is not null)
        {
            var hashed = _passwordHasher.HashPassword(instance, request.Password);
            instance.Password = hashed;
        }
        await _userRepository.UpdateAsync(instance, cancellationToken);
        return invalidated;
    }
}